app/models/landlord_record.rb

```diff
@@ -26,4 +28,8 @@ def fetch_db_connections
 
     connections
   end
+
+  # TODO: originally this should go in ApplicationRecord. By extracting this functionality (see tenant.rb:19) into this class to share across models of this gem, this doesn't work.
+  # why we need it?
+  # fetch_db_connections
 end
```

Comment:
the idea behind creating this class is that the other classes rely too much on ApplicationRecord. It's true that all apps should have an ApplicationRecord, but the reality is that it's not mandatory. By creating this intermediate class we can add stuff that both Tenant and TenantConnection use.